src/backward-link.tsx

```diff
@@ -0,0 +1,15 @@
+import type { LinkProps } from '@remix-run/react';
+import { Link } from '@remix-run/react';
+import type { To } from 'history';
+import * as React from 'react';
+import { useReturnLocation } from './hooks';
+
+export interface BackwardLinkProps extends Omit<LinkProps, 'to'> {
+  fallback?: To;
+}
+
+export function BackwardLink(props: BackwardLinkProps) {
+  const location = useReturnLocation() || props.fallback || '/';
+
+  return <Link to={location}>{props.children}</Link>;
```

Comment:
I think this should take a className prop. One thing we do a lot right now that we should stop doing is wrapping things like buttons in this case. That may have made some sense with a kit like MUI. But now that we have control over our css we should prefer instead to style our anchor tags as apposed to wrapping other styles/elements.

Reply:
I forgot to pass all of `props` into the underlying component. I meant to just extend the component, but forgot to pas props in.